rcgen/src/lib.rs

```diff
@@ -435,6 +435,25 @@ pub enum KeyUsagePurpose {
 	DecipherOnly,
 }
 
+/// Encode a key usage as the value of a BIT STRING as defined by RFC 5280.
+/// [`u16`] is sufficient to encode the largest possible key usage value (two bytes).
+impl From<&KeyUsagePurpose> for u16 {
```

Comment:
A disadvantage of using a `From` impl for this is that it instantly becomes public. As such and since we have no motivation (as far as I can tell) to expose this as public API, I'd suggest making this a `to_u16()` inherent method instead.